### PR TITLE
Feat: Backport API 6.0 to V13

### LINKED ISSUE #3004: Backport API 6.0 to v13

### PR DESCRIPTION
**Pull request by Poolitzer**
This closes #3004.

### Checklist for PRs

- [x] Added `.. versionadded:: version`, `.. versionchanged:: version` or `.. deprecated:: version` to the docstrings for user facing changes (for methods/class descriptions, arguments and attributes)
- [x] Created new or adapted existing unit tests
- [x] Documented code changes according to the [CSI standard](https://standards.mousepawmedia.com/en/stable/csi.html)
- [x] Added myself alphabetically to `AUTHORS.rst` (optional)
- [x] Added new classes & modules to the docs and all suitable `__all__` s

### If the PR contains API changes (otherwise, you can delete this passage)

* New classes:
    - [x] Added `self._id_attrs` and corresponding documentation
    - [x] `__init__` accepts `**_kwargs`
    
* Added new shortcuts:
    - [x] In `Chat` & `User` for all methods that accept `chat/user_id`
    - [ ] In `Message` for all methods that accept `chat_id` and `message_id`
    - [ ] For new `Message` shortcuts: Added `quote` argument if methods accepts `reply_to_message_id`
    - [ ] In `CallbackQuery` for all methods that accept either `chat_id` and `message_id` or `inline_message_id`
    
* If relevant:

    - [x] Added new constants at `telegram.constants` and shortcuts to them as class variables
    - [x] Link new and existing constants in docstrings instead of hard coded number and strings
    - [ ] Add new message types to `Message.effective_attachment`
    - [ ] Added new handlers for new update types
      - [ ] Add the handlers to the warning loop in the `ConversationHandler`
    - [x] Added new filters for new message (sub)types
    - [x] Added or updated documentation for the changed class(es) and/or method(s)
    - [x] Added or updated `bot_methods.rst`
    - [x] Updated the Bot API version number in all places: `README.rst` and `README_RAW.rst` (including the badge), as well as `telegram.constants.BOT_API_VERSION`
    - [ ] Added logic for arbitrary callback data in `tg.ext.Bot` for new methods that either accept a `reply_markup` in some form or have a return type that is/contains `telegram.Message`

### CODE REVIEW

File: tests/test_official.py [hunk 1]
```diff
@@ -176,6 +185,16 @@ def check_object(h4):
         ignored |= {'message', 'type', 'source'}
     elif name.startswith('InputMedia'):
         ignored |= {'filename'}  # Convenience parameter
+    elif name == 'PromoteChatMember' or name == 'ChatMemberAdministrator':
```
Comment (harshil21) [inline tests/test_official.py:188]:
``` suggestion
    elif name == 'ChatMemberAdministrator':
```
`PromoteChatMember` is not a class?

**Review by harshil21 — CHANGES_REQUESTED** [review]
Nice work! Found some small errors:

File: README.rst [hunk 2]
```diff
@@ -87,13 +89,21 @@ Table of contents
 
 - `License`_
 
+====================
+Depreciation warning
+====================
```
Comment (Poolitzer) [inline README.rst:94]:
```suggestion
Deprecation warning
====================
```

File: README.rst [hunk 3]
```diff
@@ -87,13 +89,21 @@ Table of contents
...
+
+> You are reading the documentation of V13.
+> This version is currently in "Bot API compatibility mode", so will only get updates to include new Bot API features.
+> If you can, you should switch your code to V20.
```
Comment (Poolitzer) [inline README.rst:98]:
```suggestion
    You are reading the documentation of V13. This version is currently in "Bot API compatibility mode", so will only get updates to include new Bot API features.
    
    If you can, you should switch your code to V20.
```

File: README_RAW.rst [hunk 4]
```diff
@@ -67,6 +67,8 @@ We have a vibrant community of developers helping each other in our `Telegram gr
 Table of contents
 =================
 
+- `Depreciation warning`_
```
Comment (Poolitzer) [inline README_RAW.rst:70]:
```suggestion
- `Deprecation warning`_
```

File: README_RAW.rst [hunk 5]
```diff
@@ -85,13 +87,21 @@ Table of contents
 
 - `License`_
 
+====================
+Depreciation warning
```
Comment (Poolitzer) [inline README_RAW.rst:91]:
```suggestion
Deprecation warning
```

File: tests/test_official.py (hunk 1 above)
Comment (Poolitzer) [reply tests/test_official.py:188]: right, leftover

File: telegram/webappdata.py [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+    Attributes:
+        data (:obj:`str`): The data. Be aware that a bad client can send arbitrary data in this
+            field.
+        button_text (:obj:`str`): Text of the :attr:`~telegram.KeyboardButton.web_app` keyboard
+            button, from which the Web App was opened.
+
+            Warning:
+                Be aware that a bad client can send
+            arbitrary data in this field.
```
Comment (Poolitzer) [inline telegram/webappdata.py:49]:
```suggestion
            Warning:
                Be aware that a bad client can send arbitrary data in this field.
```

[the remark this answers is not in the document]
Comment (Poolitzer) [reply telegram/webappdata.py:49]: It does not.

```suggestion
                Be aware that a bad client can send arbitrary data in this field.
```

File: telegram/message.py [hunk 7]
```diff
@@ -651,14 +762,24 @@ def de_json(cls, data: Optional[JSONDict], bot: 'Bot') -> Optional['Message']:
...
         )
         data['reply_markup'] = InlineKeyboardMarkup.de_json(data.get('reply_markup'), bot)
-        data['voice_chat_scheduled'] = VoiceChatScheduled.de_json(
+        data['voice_chat_scheduled'] = VideoChatScheduled.de_json(
             data.get('voice_chat_scheduled'), bot
         )
-        data['voice_chat_started'] = VoiceChatStarted.de_json(data.get('voice_chat_started'), bot)
-        data['voice_chat_ended'] = VoiceChatEnded.de_json(data.get('voice_chat_ended'), bot)
-        data['voice_chat_participants_invited'] = VoiceChatParticipantsInvited.de_json(
+        data['voice_chat_started'] = VideoChatStarted.de_json(data.get('vioice_chat_started'), bot)
```
[the remark this answers is not in the document]
Comment (Poolitzer) [reply telegram/message.py:768]: good catch!

File: telegram/chatadministratorrights.py [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+        .. versionadded:: 20.0
+        """
+        return cls(True, True, True, True, True, True, True, True, True, True, True)
+
+    @classmethod
+    def no_rights(cls) -> 'ChatAdministratorRights':
+        """
+        This method returns the :class:`ChatAdministratorRights` object with all attributes set to
+        :obj:`False`.
+        .. versionadded:: 20.0
```
Comment (Poolitzer) [inline telegram/chatadministratorrights.py:167]:
```suggestion
        .. versionadded:: 20.0
```

File: docs/source/telegram.menubuttoncommands.rst [hunk 9]
```diff
@@ -0,0 +1,8 @@
+:github_url: https://github.com/python-telegram-bot/python-telegram-bot/blob/v13.x/telegram/menubuttoncommands.py
```
[the remark this answers is not in the document]
Comment (Poolitzer) [reply docs/source/telegram.menubuttoncommands.rst:1]: ah missed that one

File: docs/source/telegram.menubuttondefault.rst [hunk 10]
```diff
@@ -0,0 +1,8 @@
+:github_url: https://github.com/python-telegram-bot/python-telegram-bot/blob/v13.x/telegram/menubuttondefault.py
```
Comment (Poolitzer) [inline docs/source/telegram.menubuttondefault.rst:1]:
```suggestion
:github_url: https://github.com/python-telegram-bot/python-telegram-bot/blob/v13.x/telegram/menubutton.py
```

File: docs/source/telegram.menubuttonwebapp.rst [hunk 11]
```diff
@@ -0,0 +1,8 @@
+:github_url: https://github.com/python-telegram-bot/python-telegram-bot/blob/v13.x/telegram/menubuttonwebapp.py
```
Comment (Poolitzer) [inline docs/source/telegram.menubuttonwebapp.rst:1]:
```suggestion
:github_url: https://github.com/python-telegram-bot/python-telegram-bot/blob/v13.x/telegram/menubutton.py
```

File: telegram/videochat.py [hunk 12]
```diff
@@ -0,0 +1,167 @@
...
+    This object represents a service message about a video
+    chat started in the chat. Currently holds no information.
+
+    .. versionadded:: 13.12
+    """
+
+    __slots__ = ()
+
+    def __init__(self, **_kwargs: Any):  # skipcq: PTC-W0049
+        pass
```
Comment (harshil21) [inline telegram/videochat.py:44]: `_id_attrs` missing, hence failing test

File: README.rst [hunk 13]
```diff
@@ -87,13 +89,21 @@ Table of contents
...
 - `License`_
 
+====================
+Deprecation warning
+====================
+
+    You are reading the documentation of V13. This version is currently in "Bot API compatibility mode", so will only get updates to include new Bot API features.
+    
+    If you can, you should switch your code to V20.
+
```
Comment (Bibo-Joshi) [inline README.rst:99]: can we move this to the doc-fixes-v13 pr?

File: README_RAW.rst [hunk 14]
```diff
@@ -85,13 +87,21 @@ Table of contents
 
 - `License`_
 
+====================
```
Comment (Bibo-Joshi) [inline README_RAW.rst:90]: .

File: docs/source/telegram.animation.rst [hunk 15]
```diff
@@ -1,4 +1,4 @@
-:github_url: https://github.com/python-telegram-bot/python-telegram-bot/blob/master/telegram/files/animation.py
+:github_url: https://github.com/python-telegram-bot/python-telegram-bot/blob/v13.x/telegram/files/animation.py
```
Comment (Bibo-Joshi) [inline docs/source/telegram.animation.rst:1]: same. if those changes are not individual commits, e.g. hard to revert & cherry pick, I'm also okay with keeping as is.

File: setup.py [hunk 16]
```diff
@@ -98,7 +98,6 @@ def get_setup_kwargs(raw=False):
                 'Topic :: Internet',
                 'Programming Language :: Python',
                 'Programming Language :: Python :: 3',
-                'Programming Language :: Python :: 3.6',
```
Comment (Bibo-Joshi) [inline setup.py:101]: do the changes include anything that doesn't work on v3.6? If not, then IMO we can even keep support for 3.6. on v20 we explicitly use 3.7+ code so there it makes more sense.

File: telegram/bot.py [hunk 17]
```diff
@@ -4031,6 +4085,9 @@ def promote_chat_member(
             :class:`telegram.error.TelegramError`
 
         """
+        if can_manage_voice_chats is not None and can_manage_video_chats is not None:
+            raise ValueError("Only supply `can_manage_video_chats`, not `can_manage_voice_chats`.")
+
```
Comment (Bibo-Joshi) [inline telegram/bot.py:4090]: "only supply one of …, not both"

File: telegram/chat.py [hunk 18]
```diff
@@ -600,6 +601,10 @@ def promote_member(
 
         .. versionadded:: 13.2
 
+        ..versionchanged:: 13.12
+            Since Bot API 6.0, voice chat was renamed to video chat. The old parameter is
+            supported but is supplied as video_chats.
```
Comment (Bibo-Joshi) [inline telegram/chat.py:606]: what does `is supplied as video_chats` mean? I don't understand it without reading the code

File: telegram/chatmember.py [hunk 19]
```diff
@@ -327,8 +328,17 @@ def __init__(
...
+        can_manage_video_chats: bool = None,
         **_kwargs: Any,
     ):
+        # check before required to not waste resources if the error is raised
+        if can_manage_voice_chats is not None and can_manage_video_chats is not None:
+            # if they are the same it's fine...
+            if can_manage_voice_chats != can_manage_video_chats:
+                raise ValueError(
+                    "Only supply `can_manage_video_chats`, " "not `can_manage_voice_chats`."
+                )
```
Comment (Bibo-Joshi) [inline telegram/chatmember.py:340]: see above

File: telegram/chatmember.py [hunk 20]
```diff
@@ -456,6 +475,12 @@ class ChatMemberAdministrator(ChatMember):
             administrator can delete messages of other users.
         can_manage_voice_chats (:obj:`bool`, optional): :obj:`True`, if the
             administrator can manage voice chats.
+
+            .. deprecated:: 13.12
```
Comment (Bibo-Joshi) [inline telegram/chatmember.py:479]: "Contains the same value as *_voice_* for backwards compatibility" or similar wording. same everywhere, where applicable

File: telegram/message.py [hunk 21]
```diff
@@ -88,6 +93,10 @@ class Message(TelegramObject):
     Note:
         In Python ``from`` is a reserved word, use ``from_user`` instead.
 
+    .. versionchanged:: 13.12
+        Since Bot API 6.0, voice chat was renamed to video chat. The old parameter is
+        supported but is supplied as video_chats.
```
Comment (Bibo-Joshi) [inline telegram/message.py:98]: see above

File: telegram/message.py [hunk 22]
```diff
@@ -516,8 +577,40 @@ def __init__(
...
+        video_chat_participants_invited: VideoChatParticipantsInvited = None,
+        web_app_data: WebAppData = None,
         **_kwargs: Any,
     ):
+        if (
+            voice_chat_scheduled is not None
+            and video_chat_scheduled is not None
+            and voice_chat_scheduled != video_chat_scheduled
+        ):
+            raise ValueError("Only supply `video_chat_scheduled`, not `voice_chat_scheduled`.")
```
Comment (Bibo-Joshi) [inline telegram/message.py:592]: sa

**Review by Bibo-Joshi — COMMENTED** [review]
great work 👍  some nitpicking below

about pre-commit.ci: we can either tell it to recurse submodules (we'd have to change that on master, I guess) or add something like `[skip pre-commit.ci]` to the commit messages on v13.x.

File: README.rst (hunk 13 above)
Comment (Poolitzer) [reply README.rst:99]: I can rollback the changes sure

File: setup.py (hunk 16 above)
Comment (Poolitzer) [reply setup.py:101]: yes, cryptography fails.

Comment (Bibo-Joshi) [reply setup.py:101]: fair enough

File: telegram/chat.py (hunk 18 above)
Comment (Poolitzer) [reply telegram/chat.py:606]: https://github.com/python-telegram-bot/python-telegram-bot/pull/3027#discussion_r871313141

Harshil suggested to drop that part completely in a different file, can do here as well

Comment (Bibo-Joshi) [reply telegram/chat.py:606]: also fine with me 👍

**Review by Bibo-Joshi — COMMENTED** [review]
pre-commit is failing because there's a py36 related workaround still in `ext.__init__`, please remove that. otherwise LGTM from my side.

**Comment by Bibo-Joshi** [comment]
ps: `[skip pre-commit.ci]` apparently has no effect? what's up with deppsource?

**Comment by Poolitzer** [comment]
@Bibo-Joshi Both of them unrelated to my changes I think.

File: telegram/ext/__init__.py [hunk 23]
```diff
@@ -28,17 +28,6 @@
 from .contexttypes import ContextTypes
 from .dispatcher import Dispatcher, DispatcherHandlerStop, run_async
 
-# https://bugs.python.org/issue41451, fixed on 3.7+, doesn't actually remove slots
-# try-except is just  here in case the __init__ is called twice (like in the tests)
-# this block is also the reason for the pylint-ignore at the top of the file
```
Comment (Bibo-Joshi) [inline telegram/ext/__init__.py:33]: can you also remove the pylint-ignores?

**Review by Bibo-Joshi — COMMENTED** [review]
I just see that codecov marks some of eth "mutually exclusive" tests as uncovered - can you add corresponding tests?

I told deepsource to ignore the 3 issues. for pre-commit, can you try one of the other [documented comments](https://pre-commit.ci) and/or putting it into the first line of the commit message? If neither of that works, I'll probably change the submodules setting on master …

File: telegram/ext/__init__.py (hunk 23 above)
Comment (harshil21) [reply telegram/ext/__init__.py:33]: we can also remove all py 3.7 workarounds (if there were more?)

Comment (Bibo-Joshi) [reply telegram/ext/__init__.py:33]: where there? I just found one in test_ppersistence and I'm fine with leaving that …

File: tests/test_message.py [hunk 24]
```diff
@@ -1604,3 +1604,31 @@ def test_equality(self):
...
+                video_chat_ended=False
+            )
+            Message(
+                self.id_,
+                self.date,
+                self.chat,
+                from_user=self.from_user,
+                voice_chat_participants_invited=True,
+                video_chat_participants_invited=False
+            )
```
Comment (harshil21) [inline tests/test_message.py:1633]: these should have their own `with` block

File: tests/test_chatmember.py [hunk 25]
```diff
@@ -255,3 +255,7 @@ def test_equality(self, chat_member_types, user):
 
         assert c != e
         assert hash(c) != hash(e)
+
+    def test_invalid_input(self, user):
+        with pytest.raises(ValueError):
+            ChatMember(user=user, status="status", can_manage_video_chats=True, can_manage_voice_chats=False)
```
Comment (Bibo-Joshi) [inline tests/test_chatmember.py:261]: tiny one: missing new line.

**Review by Bibo-Joshi — COMMENTED** [review]
> for pre-commit, can you try one of the other [documented comments](https://pre-commit.ci) and/or putting it into the first line of the commit message? If neither of that works, I'll probably change the submodules setting on master …

**Pull request merged**